oragono.go

```diff
@@ -71,7 +71,7 @@ Options:
 			log.Println("database initialized: ", config.Datastore.Path)
 		}
 	} else if arguments["upgradedb"].(bool) {
-		irc.UpgradeDB(config.Datastore.Path)
+		irc.UpgradeDB(config)
```

Comment:
Hmm, I'm honestly a bit iffy about this thing wholesale. Are we able to make this easier by giving the `UpgradeDB()` function:

- a `buntdb.DB` object directly
- a struct filled with data that's used by upgrade (e.g. `data.DefaultChannelModes`, etc)

I think changing this function to accept those arguments should also make it possible to write proper tests for our UpgradeDB function and ensure it doesn't miss or lose anything as we write new stuff. I'd like to, for example, have a test that produces an example `v1` database, upgrades it to the latest and then confirms along the way that everything got upgraded properly.

Reply:
1. I think it's not ideal for `oragono.go` to be directly aware of the config data that's needed to perform individual schema changes (it could get messy).
1. It also seems like the use of `Config` here isn't a big obstacle to writing tests, because it shouldn't force us to write YAML in the test or anything; we can just create a zero-valued `Config` struct, manually initialize whatever parts are relevant to the schema change
1. It seems likely that (for example) by the time we get to v5, we won't actually have code support for reconstructing a v1 database (it would require us to retain buggy code paths); it seems like the best we could do is store "canned" databases from various old versions.

Reply:
Hmm... yeah I can see something like that. In that case yeah I'd be alright leaving it as-is for now, and we can always clean it up more later.